src/components/smartSearch/filters/CampaignParticipation/DisplayCampaignParticipation.tsx

```diff
@@ -0,0 +1,73 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { useQuery } from 'react-query';
+import { useRouter } from 'next/router';
+
+import getActivity from 'fetching/getActivity';
+import getCampaign from 'fetching/getCampaign';
+import getLocation from 'fetching/getLocation';
+import { getTimeFrameWithConfig } from '../../utils';
+import { CampaignParticipationConfig, OPERATION, SmartSearchFilterWithId } from 'types/smartSearch';
+
+interface DisplayCampaignParticipationProps {
+    filter: SmartSearchFilterWithId<CampaignParticipationConfig>;
+}
+
+const DisplayCampaignParticipation = ({ filter }: DisplayCampaignParticipationProps) : JSX.Element => {
+    const { orgId } = useRouter().query;
+    const { config } = filter;
+    const { operator, state, campaign: campId, activity: activityId, location: locationId } = config;
+    const op = filter.op || OPERATION.ADD;
+    const { timeFrame, after, before, numDays } = getTimeFrameWithConfig({ after: config.after, before: config.before });
+
+    const campaignQuery = useQuery(['campaign', orgId, campId ], getCampaign(orgId as string, campId?.toString() as string), { enabled: !!campId });
+    const activityQuery = useQuery(['activity', orgId, activityId ], getActivity(orgId as string, activityId?.toString() as string), { enabled: !!activityId });
+    const locationQuery = useQuery(['campaign', orgId, locationId ], getLocation(orgId as string, locationId?.toString() as string), { enabled: !!locationId });
```

Comment:
@richardolsson what do you think about making these queries individually as opposed to getting all the locations / activities / campaigns and then filtering them each time to get the correct title? Which operation is more expensive?

Reply:
I think they are fine like this. What we need to do at some point however is establish a pattern for naming cache keys, and invalidating cache keys at certain points. For example, if we store individual locations etc (like this) in the cache, then they could potentially go out of sync with the same location objects in a list of locations also stored separately in the cache.

But that's for another day, and won't be a problem generally because of react-querys automatic cache invalidation/refresh (e.g. on window activation).

Reply:
ouch, I see a typo!

Reply:
I see another (more complex) way of doing ts object destruction here! Happy to be able to understand it now, @richardolsson 🎉🙌

Reply:
There is no object destructoring happening on these lines, as far as I can tell. I think maybe what you're seeing is the `{ enabled: !!campId }` etc, but that's just creating (constructing) an object using plain old javascript, like `{ foo: 1 }`.

Maybe the `!!` is confusing you? It's actually just a way of converting something into a boolean from any [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) or [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value.

All values in javascript are either truthy or falsy. Only boolean variables are exactly `true` or `false`, but all other values as well lean more towards true (truthy) or false (falsy). For example, `null` is falsy and 1 is truthy. If you use `null` in a context where it will be interpreted as a boolean (e.g. in an if statement) it will be interpreted as `false`.

The `!` operator means "not". It interprets it's operand as a boolean and resolves to the opposite boolean, i.e. `!false` means `true` and `!true` means `false`. Because `!` only works on booleans, the operand is converted to a boolean, which is where being "truthy" or "falsy" comes in.

If something is falsy it means that when interpreted as a boolean, it will be interpreted as `false`. One example is `null`, so `!null` will be interpreted as `true` (_**not** false_).

If `!false` is `true` and `!true` is `false` it must consequently mean that `!!false` is `false` through double negation: It's **NOT NOT** false. That's what the double `!!` means. It converts something to boolean while negating it, and then negates it again to go back to the original value but always as a boolean (while the original value can be anything that is either truthy or falsy).

It's not a special operator, it's just a very common operator (`!`) twice.

Reply:
Oops! Yeah, now that I see it more carefully I understand what I confused it for. Very thorough! Thank you @richardolsson! 🙏